### PR TITLE
Barrier clean

### PR DESCRIPTION
**Pull request by jiyess**
This branch features our construction of analysis-suitable parameterizations from input boundary representations, comprising the following five files:
- `gsAnalysisSuitableParameterization.cpp`: An example file demonstrating usage.
- `gsBarrierPatch.h`: Header file for the Barrier Patch class.
- `gsBarrierPatch.hpp`: Implementation file for the Barrier Patch class.
- `gsBarrierCore.h`: Header file for the Barrier Core class.
- `gsBarrierCore.hpp`: Implementation file for the Barrier Core class.

In the folder `/filedata/breps/`, we've included various test files for your convenience. Please note that redundant files in the `/filedata` folder have been cleared out to maintain cleanliness and organization.

Pull requests can only be merged after at least one review (and
approval) from @gismo/admins.

Code submitted to the stable branch should be clean, well-documented
and free of bugs.

# Please consider the following checklist before issuing a pull
request:
- [X] Have you added an explanation of what your changes do and why
  you'd like us to include them?
- [X] Have you documented any new codes using Doxygen comments?
- [X] Have you written new tests or examples for your changes?
-----

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -242,6 +242,10 @@ if(GISMO_WITH_UMFPACK)
   #include_directories(SYSTEM ${UMFPACK_INCLUDES})
 endif(GISMO_WITH_UMFPACK)
 
+if(GISMO_WITH_HLBFGS)
```
Comment (filiatra) [inline CMakeLists.txt]: This is not needed, or at least it should not be needed.

File: examples/gsAnalysisSuitableParameterization.cpp
Comment (filiatra) [inline examples/gsAnalysisSuitableParameterization.cpp:1]: Please rename as 
an_parameterization_example.cpp

File: src/gsAssembler/gsExpressions.h [hunk 2]
```diff
@@ -2380,6 +2388,39 @@ class sign_expr : public _expr<sign_expr<E> >
     void print(std::ostream &os) const { os<<"sgn("; _u.print(os); os <<")"; }
 };
 
+/**
+   Expression for the exponent of expression
```
Comment (filiatra) [inline src/gsAssembler/gsExpressions.h]: rename as
exp_expression 

I think the more correct term is exponential

File: src/gsAssembler/gsExpressions.h [hunk 3]
```diff
@@ -2415,6 +2456,36 @@ class ppart_expr : public _expr<ppart_expr<E> >
     void print(std::ostream &os) const { os<<"posPart("; _u.print(os); os <<")"; }
 };
 
+template<class E>
```
Comment (filiatra) [inline src/gsAssembler/gsExpressions.h:2462]: please add some words of documentation.

**Review by filiatra — CHANGES_REQUESTED** [review]

File: src/gsModeling/gsBarrierPatch.h [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): Ye Ji, H.M. Verhelst
+*/
+
+#pragma once
+
+#include <gismo.h>
+#include <gsHLBFGS/gsHLBFGS.h>
```
Comment (hverhelst) [inline src/gsModeling/gsBarrierPatch.h:19]: Maybe you need
```
#ifdef gsHLBFGS_ENABLED
```
Maybe you need
```
#ifdef gsHLBFGS_ENABLED
#include <gsHLBFGS/gsHLBFGS.h>
#endif
```
but probably the include can be completely removed.
#include <gsHLBFGS/gsHLBFGS.h>
#endif
```
but probably the include can be completely removed.

File: src/gsModeling/gsBarrierCore.h [hunk 5]
```diff
@@ -0,0 +1,853 @@
...
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): Ye Ji, H.M. Verhelst
+*/
+
+#pragma once
+
+#include <gismo.h>
+#include <gsHLBFGS/gsHLBFGS.h>
```
Comment (hverhelst) [inline src/gsModeling/gsBarrierCore.h:24]: Maybe you need
```
#ifdef gsHLBFGS_ENABLEDMaybe you need
```
#ifdef gsHLBFGS_ENABLED
#include <gsHLBFGS/gsHLBFGS.h>
#endif
```
but probably the include can be completely removed.
#include <gsHLBFGS/gsHLBFGS.h>
#endif
```
Also think about the case where HLBFGS is not loaded (e.g. use gradient-based algorithm)

**Review by filiatra — APPROVED** [review]

**Pull request merged**